ivy/functional/frontends/numpy/ndarray/ndarray.py

```diff
@@ -249,10 +249,15 @@ def transpose(self, axes, /):
     def swapaxes(self, axis1, axis2, /):
         return np_frontend.swapaxes(self, axis1, axis2)
 
-    def all(self, axis=None, out=None, keepdims=False, *, where=True):
+    def all(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
         return np_frontend.all(self, axis, out, keepdims, where=where)
 
-    def any(self, axis=None, out=None, keepdims=False, *, where=True):
+    def any(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
+        if not (dtype is None or ivy.is_bool_dtype(dtype)):
+            raise TypeError(
+                "No loop matching the specified signature and "
+                "casting was found for ufunc logical_or"
+            )
```

Comment:
I think something like this should be included @DragosStoican

Reply:
ok, makes sense, I had a look at np implementation too and I think in this case the addition of dtype arg is justified. Can you make sure this error is consistent across both `any` and `all` functions?

Reply:
Done, can you check now @DragosStoican